stix2/equivalence/graph_equivalence/__init__.py

```diff
@@ -0,0 +1,209 @@
+        weight_dict: A dictionary that can be used to override settings
+            in the semantic equivalence process
+
+    Returns:
+        float: A number between 0.0 and 100.0 as a measurement of equivalence.
+
+    Warning:
+        Some object types do not have an entry for use in the equivalence process.
+        In order for those objects to influence the final score a new entry needs to
+        be defined in the WEIGHTS dictionary. Similarly, the values can be fine tuned
+        for a particular use case. Graph equivalence has additional entries. The
+        complete graph is needed for the two graphs that are being checked.
+
+    Note:
+        Default weights_dict:
+
+        .. include:: ../default_sem_eq_weights.rst
+
+    """
+    weights = WEIGHTS.copy()
+
+    if weight_dict:
+        weights.update(weight_dict)
+
+    results = {}
+    depth = weights["_internal"]["max_depth"]
+
+    graph1 = ds1.query([])
+    graph2 = ds2.query([])
+
+    graph1.sort(key=lambda x: x["type"])
+    graph2.sort(key=lambda x: x["type"])
+
+    if len(graph1) < len(graph2):
+        weights["_internal"]["ds1"] = ds1
+        weights["_internal"]["ds2"] = ds2
+        g1 = graph1
+        g2 = graph2
+    else:
+        weights["_internal"]["ds1"] = ds2
+        weights["_internal"]["ds2"] = ds1
+        g1 = graph2
+        g2 = graph1
+
+    for object1 in g1:
+        for object2 in g2:
+            if object1["type"] == object2["type"]:
+                iprop_score = {}
+                result = semantically_equivalent(object1, object2, iprop_score, **weights)
+                objects1_id = object1["id"]
+                weights["_internal"]["max_depth"] = depth
+
+                if objects1_id not in results:
+                    results[objects1_id] = {"matched": object2["id"], "prop_score": iprop_score, "value": result}
+                elif result > results[objects1_id]["value"]:
+                    results[objects1_id] = {"matched": object2["id"], "prop_score": iprop_score, "value": result}
+
+    matching_score = sum(x["value"] for x in results.values())
+    sum_weights = len(g1) * 100.0
+    equivalence_score = (matching_score / sum_weights) * 100
+    prop_scores["matching_score"] = matching_score
+    prop_scores["sum_weights"] = sum_weights
+    prop_scores["summary"] = results
+
+    logger.debug(
+        "DONE\t\tSUM_WEIGHT: %.2f\tMATCHING_SCORE: %.2f\t SCORE: %.2f",
+        sum_weights,
+        matching_score,
+        equivalence_score,
+    )
+    return equivalence_score
+
+
+# default weights used for the graph semantic equivalence process
+# values are re-balanced to account for new property checks and add up to 100
+WEIGHTS = {
+    "attack-pattern": {
+        "name": (30, partial_string_based),
+        "external_references": (60, partial_external_reference_based),
+        "created_by_ref": (5, semantic_check),
+        "object_marking_refs": (5, list_semantic_check),
```

Comment:
As discussed offline, let's remove `created_by_ref` and `object_marking_refs` from the calculation since two organizations might create what is semantically the same content, or that same content might get marked differently by different organizations.